### PR TITLE
Background layers processor

### PR DESCRIPTION
**Pull request by jesseeichar**
Added Processor for adding background layers to a map. See #231 for details.

### CODE REVIEW

**Comment by tsauerwein** [comment]
I added some comments, but this is looking good!

Will this PR also close #230? Or do you still want something generic (for what use-case)?

**Comment by sbrunner** [comment]
Looks good but shouldn't he also be able to add overlay ?
Sometimes we want to add a graticule, watermark, or something else generated by the WMS server, ...

**Comment by jesseeichar** [comment]
I have updated this PR so that it can do overlays or background layers.  But unfortunately I rebased onto develop and lost the comments of @tsauerwein 

@tsauerwein do you remember your comments?  Or perhaps you can re-review this?  I need this for the sogelink.

File: core/src/main/java/org/mapfish/print/processor/map/AddStaticLayersProcessor.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+        /**
+         * The map to update with the static layers.
+         */
+        public MapAttributeValues map;
+
+        /**
+         * The attribute containing the static layers to add to the map.
+         */
+        public StaticLayersAttribute.StaticLayersAttributeValue staticLayers;
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/processor/map/AddStaticLayersProcessor.java]: `backgroundLayers` isn't better than `staticLayers`? (for the attribute name)

Comment (jesseeichar) [inline core/src/main/java/org/mapfish/print/processor/map/AddStaticLayersProcessor.java]: It is not necessarily background layers.  It could be overlay layers

Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/processor/map/AddStaticLayersProcessor.java]: OK, thanks :-)

File: core/src/main/java/org/mapfish/print/processor/map/AddStaticLayersProcessor.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import org.mapfish.print.wrapper.PJoinedArray;
+
+import java.util.Arrays;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * This processor adds the configured set of layers to the map.
+ *
+ * This is useful when all maps should have a default set of background layers or overlayers
```
Comment (tsauerwein) [inline core/src/main/java/org/mapfish/print/processor/map/AddStaticLayersProcessor.java]: "... overlays"

File: core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    @Override
+    public int size() {
+        return this.combinedSize;
+    }
+
+    @Override
+    public PObject getObject(final int i) {
+        int index = i;
```
Comment (tsauerwein) [inline core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java]: Could you also use `get()` for this method? Same for `getArray`.

Comment (jesseeichar) [inline core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java]: That would require an API change of an API that goes back to V1.  I'd rather leave the API alone.  Or if we want to change it we can do it in a separate PR.

Comment (tsauerwein) [inline core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java]: Maybe I am missing something, but wouldn't `return (PObject) get(i);` do the same thing?

**Comment by tsauerwein** [comment]
My original comment (which was lost) was about the dependencies in `mapfish-spring-processors.xml`, which you addressed already. So it looks good to me.

File: core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java (hunk 3 above)
Comment (jesseeichar) [inline core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java]: Ah,  I misunderstood the question.  I think in some cases it will not return the same thing but I am not sure.  Its worth looking into a bit more.
